File: 백준/Silver/20310. 타노스/타노스.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String input = br.readLine().trim();

        ArrayList<Character> S = new ArrayList<>();

        for(char c : input.toCharArray()) {
            S.add(c);
        }

        long zeroCount = S.stream().filter(c -> c == '0').count();
        long oneCount = S.stream().filter(c -> c == '1').count();

        int stopPoint = 0;
        for(int i = 0; i < S.size(); i++) {
            if(S.get(i) == '1') {
                S.remove(i);
                i--;
                stopPoint++;
            }

            if(stopPoint == oneCount/2) {
                break;
            }
        }


        stopPoint = 0;
        for(int i = S.size()-1; i >= 0; i--) {
            if(S.get(i) == '0') {
                S.remove(i);
                stopPoint++;
            }

            if(stopPoint == zeroCount/2) {
                break;
            }
        }

        StringBuilder result = new StringBuilder();

        for(char c : S) {
            result.append(c);
        }

        System.out.println(result.toString());
    }
}

File: 백준/Silver/20310. 타노스/타노스.py
import sys
input = sys.stdin.readline

S = list(input().strip())

zero_count = S.count('0')
one_count = S.count('1')

stop_point = 0
for w in S:
    if w == '1':
        S.remove(w)
        stop_point += 1

    if stop_point == one_count//2:
        break

S = S[::-1]
stop_point = 0
for w in S:
    if w == '0':
        S.remove(w)
        stop_point += 1

    if stop_point == zero_count // 2:
        break

print(''.join(S[::-1]))
